packages/quantic/force-app/main/default/lwc/quanticGeneratedAnswer/quanticGeneratedAnswer.js

```diff
@@ -286,9 +292,21 @@ export default class QuanticGeneratedAnswer extends LightningElement {
    * handles liking the generated answer.
    * @param {CustomEvent} event
    */
-  handleLike(event) {
+  async handleLike(event) {
     event.stopPropagation();
-    this.generatedAnswer.like?.();
+    if (!this.feedbackSubmitted && !this._liked) {
+      this._liked = true;
+      this._disliked = false;
+      this.generatedAnswer.like?.();
+      // @ts-ignore
+      await FeedbackModalQna.open({
+        size: 'small',
+        liked: true,
+        label: this.labels.feedbackHelpUsImprove,
+        handleSubmit: this.submitFeedback.bind(this),
+      });
```

Comment:
```suggestion
      await FeedbackModalQna.open({
        size: 'small',
        label: this.labels.feedbackHelpUsImprove,
        handleSubmit: (payload) => {this.submitFeedback(...payload, helpful: true)},
      });
```
you may also need to transform `submitFeedback` to an arrow function to avoid the `bind(this)`

Reply:
Interesting, this implies the feedback modal doesn't know if it was triggered by like or dislike